### PR TITLE
Examples of Using Secret Template

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

Changes proposed in this pull request:

- an example of how to use secret template with a new release of [sap-btp-operator-service](https://github.com/sap/sap-btp-service-operator/releases/tag/v0.6.1)

**Related issue(s)**

### CODE REVIEW

File: docs/user/02-11-examples.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Examples
+
+The following document contains description of notable features shipped with new SAP BTP Operator Service upgrades discovered during application of newest versions.
```
Comment (ralikio) [inline docs/user/02-11-examples.md:3]: This name intended to point to https://github.com/SAP/sap-btp-service-operator

File: docs/user/02-11-examples.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Secret Templates
+
+Version 0.6.1 introduced secret templates feature that allows to modify contents of the secret that is normally generated from `ServiceBinding` instance created for any service. The customer can now specify a go template as a value of `secretTemplate` field inside of binding's spec section. Inside of the template you can refer (by using `{{}}` syntax) to credentials stored inside of a service binding (with `{{credentials.<key>}}`) or information of given instance (with `{{instance.<key>}}`). Parameter that can be used with `instance` key are limited to the values set in [getInstanceInfo](https://github.com/SAP/sap-btp-service-operator/blob/8c0a3d7d7ca54e44143c0e0b7d1e1ef206b362ab/controllers/servicebinding_controller.go#L819) method. Below is an example of a service binding with spectTemplate value field:
```
Comment (IwonaLanger) [inline docs/user/02-11-examples.md:7]: 1. Please capitalize "secret" if referring to a Kubernetes Secret.
2. Is "spec**t**Template" correct? Shouldn't it be "secretTemplate"? Or perhaps "specTemplate"?

File: docs/user/02-11-examples.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Examples
```
Comment (IwonaLanger) [inline docs/user/02-11-examples.md:1]: Can you make the title more descriptive?

Comment (ralikio) [reply docs/user/02-11-examples.md]: Corrected to "Sap BTP Service Operator Examples".

File: docs/user/02-11-examples.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ralikio) [reply docs/user/02-11-examples.md]: Corrected with modifications for articles of `the secret's content` and `the ServiceBindng`.

Comment (ralikio) [reply docs/user/02-11-examples.md]: Both corrected. Thank you!

File: docs/user/02-11-examples.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Sap BTP Service Operator Examples
+
+This document describes notable features shipped with new SAP BTP service operator upgrades discovered during the application of latest versions.
+
+## Secret Templates
+
+Version 0.6.1 introduced a Secret templates feature that allows for modification of Secret's content that is normally generated from a ServiceBinding instance created for any service. You can now specify a Go template as a value of the **secretTemplate** field inside the binding's spec section. Inside the template, you can refer (by using `{{}}` syntax) to credentials stored inside of a ServiceBinding (with `{{credentials.<key>}}`) or information of a given instance (with `{{instance.<key>}}`). The parameters that can be used with the `instance` key are limited to the values set in the [getInstanceInfo](https://github.com/SAP/sap-btp-service-operator/blob/8c0a3d7d7ca54e44143c0e0b7d1e1ef206b362ab/controllers/servicebinding_controller.go#L819) method. Here is an example of a ServiceBinding with the **secretTemplate** field:
```
Comment (jaroslaw-pieszka) [inline docs/user/02-11-examples.md:7]: Reference to particular line number could be unfortunate.

Comment (jaroslaw-pieszka) [inline docs/user/02-11-examples.md:7]: Is it really a `Go template`?

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**